### DISCUSSION TITLE
Custom Controlled Switch triggers unexpected save onClick

### DISCUSSION THREAD

**Discussion by cdedreuille** [react-hook-form/react-hook-form#2477] [Q&A]
Hi 👋 

I 'me creating a set of custom components including a Switch that should return true or false. I made it using a Controller and it works but for some reasons it automatically save the entire form every time I click on it and I can't figure out why.

**I created a sandbox here:**
https://codesandbox.io/s/react-hook-form-switch-xww12?file=/src/index.js

**How to reproduce:**
If you click on the switch button you should see that it trigger the onSubmit function and change the button name.

Is this the best way to create a switch? Did I miss something?

**Comment by bluebill1049** [comment]
probably missing type="button"?
